File: framework/web/src/main/java/top/bettercode/simpleframework/web/serializer/annotation/JsonHide.java
package top.bettercode.simpleframework.web.serializer.annotation;

import com.fasterxml.jackson.annotation.JacksonAnnotationsInside;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import top.bettercode.simpleframework.web.serializer.HideSerializer;

/**
 * @author Peter Wu
 * @since 0.1.15
 */
@Target({ElementType.METHOD, ElementType.FIELD, ElementType.ANNOTATION_TYPE})
@Retention(RetentionPolicy.RUNTIME)
@JacksonAnnotationsInside
@Inherited
@JsonSerialize(using = HideSerializer.class)
public @interface JsonHide {

  int beginKeep() default 0;

  int endKeep() default 0;

  boolean alwaysHide() default true;
}

File: web/src/main/kotlin/top/bettercode/summer/web/serializer/annotation/JsonHide.kt
package top.bettercode.summer.web.serializer.annotation

import com.fasterxml.jackson.annotation.JacksonAnnotationsInside
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import top.bettercode.summer.web.serializer.HideSerializer
import java.lang.annotation.Inherited

/**
 * @author Peter Wu
 * @since 0.1.15
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.FIELD, AnnotationTarget.ANNOTATION_CLASS)
@Retention(AnnotationRetention.RUNTIME)
@JacksonAnnotationsInside
@Inherited
@JsonSerialize(using = HideSerializer::class)
annotation class JsonHide(val beginKeep: Int = 0, val endKeep: Int = 0, val alwaysHide: Boolean = true)
